diff --git a/acinclude.m4 b/acinclude.m4
index 51cf130..7853fd8 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -206,7 +206,7 @@ EOF
       modpath_shared="$modpath_shared $libname"
       cat >>$modpath_current/modules.mk<<EOF
 $libname: $shobjects
-	\$(SH_LINK) -rpath \$(libexecdir) -module -avoid-version $4 $objects $5
+	\$(SH_LINK) -rpath \$(libexecdir) -Wl,--unresolved-symbols=ignore-all -module -avoid-version $4 $objects $5
 EOF
     fi
   fi
@@ -260,7 +260,7 @@ EOF
             shobjects=`echo $objects | sed 's/\.lo/.slo/g'`
             cat >$mpmpath/modules.mk<<EOF
 $libname: $shobjects
-	\$(SH_LINK) -rpath \$(libexecdir) -module -avoid-version $objects $6
+	\$(SH_LINK) -rpath \$(libexecdir) -Wl,--unresolved-symbols=ignore-all -module -avoid-version $objects $6
 DISTCLEAN_TARGETS = modules.mk
 static =
 shared = $libname
diff --git a/support/apxs.in b/support/apxs.in
index 13322ca..5c58509 100644
--- a/support/apxs.in
+++ b/support/apxs.in
@@ -476,7 +476,7 @@ if ($opt_c) {
     else {
         my $apr_ldflags=`$apr_config --ldflags`;
         chomp($apr_ldflags);
-        $opt .= " -rpath $CFG_LIBEXECDIR -module -avoid-version $apr_ldflags";
+        $opt .= " -rpath $CFG_LIBEXECDIR -Wl,--unresolved-symbols=ignore-all -module -avoid-version $apr_ldflags";
     }
 
     push(@cmds, "$libtool $ltflags --mode=link $CFG_CC $ldflags -o $dso_file $opt $lo");
@@ -722,7 +722,7 @@ stop:
 
 -=#=-
 mod_%NAME%.la: mod_%NAME%.slo
-	$(SH_LINK) -rpath $(libexecdir) -module -avoid-version  mod_%NAME%.lo
+	$(SH_LINK) -rpath $(libexecdir) -Wl,--unresolved-symbols=ignore-all -module -avoid-version  mod_%NAME%.lo
 DISTCLEAN_TARGETS = modules.mk
 shared =  mod_%NAME%.la
 -=#=-
