diff --git a/configure.in b/configure.in
index 4344b07..2b185dd 100644
--- a/configure.in
+++ b/configure.in
@@ -338,14 +338,7 @@ case $host in
       APR_SETVAR(SINGLE_LISTEN_UNSERIALIZED_ACCEPT, [1])
       ;;
   *-linux-*)
-      case `uname -r` in
-        # Unserialized accept() was not recommended until Linux 2.2.
-        [[01]].* | 2.[[01]]* )
-            ;;
-        * )
-            APR_SETVAR(SINGLE_LISTEN_UNSERIALIZED_ACCEPT, [1])
-            ;;
-      esac
+      APR_SETVAR(SINGLE_LISTEN_UNSERIALIZED_ACCEPT, [1])
       ;;
   *486-*-bsdi* | *-netbsd* | *-freebsd* | *-apple-darwin* | *-dec-osf* | *-qnx)
       APR_SETVAR(SINGLE_LISTEN_UNSERIALIZED_ACCEPT, [1])
