--- support/apxs.in	2010-06-22 19:12:44.000000000 +0200
+++ support/apxs.in.oden	2012-03-03 12:52:46.884185322 +0100
@@ -450,11 +450,11 @@ if ($opt_c) {
 
     if ($opt_p == 1) {
         
-        my $apr_libs=`$apr_config --cflags --ldflags --link-libtool --libs`;
+        my $apr_libs=`$apr_config --cflags --ldflags --link-libtool`;
         chomp($apr_libs);
         my $apu_libs="";
         if ($apr_major_version < 2) {
-            $apu_libs=`$apu_config --ldflags --link-libtool --libs`;
+            $apu_libs=`$apu_config --ldflags --link-libtool`;
             chomp($apu_libs);
         }
         
