diff --git a/support/apxs.in b/support/apxs.in
index b2705fa..13322ca 100644
--- a/support/apxs.in
+++ b/support/apxs.in
@@ -463,11 +463,11 @@ if ($opt_c) {
     my $ldflags = "$CFG_LDFLAGS";
     if ($opt_p == 1) {
         
-        my $apr_libs=`$apr_config --cflags --ldflags --link-libtool --libs`;
+        my $apr_libs=`$apr_config --cflags --ldflags --link-libtool`;
         chomp($apr_libs);
         my $apu_libs="";
         if ($apr_major_version < 2) {
-            $apu_libs=`$apu_config --ldflags --link-libtool --libs`;
+            $apu_libs=`$apu_config --ldflags --link-libtool`;
             chomp($apu_libs);
         }
         
