diff --git a/server/request.c b/server/request.c
index 5599b2c..983e183 100644
--- a/server/request.c
+++ b/server/request.c
@@ -1188,12 +1188,20 @@ AP_DECLARE(int) ap_directory_walk(request_rec *r)
 #ifdef CASE_BLIND_FILESYSTEM
                 && (filename_len <= canonical_len)
 #endif
-                && ((opts.opts & (OPT_SYM_OWNER | OPT_SYM_LINKS)) == OPT_SYM_LINKS))
+                && (opts.opts & OPT_SYM_LINKS) )
             {
-
-                thisinfo.filetype = APR_DIR;
-                ++seg;
-                continue;
+                if ((r->path_info && *r->path_info) || r->finfo.filetype == APR_DIR)
+                {
+                    thisinfo.filetype = APR_DIR;
+                    ++seg;
+                    continue;
+                }
+                else if (r->finfo.filetype == APR_REG)
+                {
+                    thisinfo.filetype = APR_REG;
+                    ++seg;
+                    break;
+                }
             }
 
             /* We choose apr_stat with flag APR_FINFO_LINK here, rather that
