diff -Naur httpd-2.4.1/acinclude.m4 httpd-2.4.1.oden/acinclude.m4
--- httpd-2.4.1/acinclude.m4	2012-01-22 13:34:56.000000000 -0500
+++ httpd-2.4.1.oden/acinclude.m4	2012-03-05 04:52:38.877213712 -0500
@@ -195,7 +195,7 @@
       modpath_shared="$modpath_shared $libname"
       cat >>$modpath_current/modules.mk<<EOF
 $libname: $shobjects
-	\$(SH_LINK) -rpath \$(libexecdir) -module -avoid-version $4 $objects $5
+	\$(SH_LINK) -rpath \$(libexecdir) -Wl,--unresolved-symbols=ignore-all -module -avoid-version $4 $objects $5
 EOF
     fi
   fi
@@ -249,7 +249,7 @@
             shobjects=`echo $objects | sed 's/\.lo/.slo/g'`
             cat >$mpmpath/modules.mk<<EOF
 $libname: $shobjects
-	\$(SH_LINK) -rpath \$(libexecdir) -module -avoid-version $objects $6
+	\$(SH_LINK) -rpath \$(libexecdir) -Wl,--unresolved-symbols=ignore-all -module -avoid-version $objects $6
 DISTCLEAN_TARGETS = modules.mk
 static =
 shared = $libname
diff -Naur httpd-2.4.1/support/apxs.in httpd-2.4.1.oden/support/apxs.in
--- httpd-2.4.1/support/apxs.in	2012-03-05 04:53:08.201217284 -0500
+++ httpd-2.4.1.oden/support/apxs.in	2012-03-05 04:52:54.943213428 -0500
@@ -463,7 +463,7 @@
     else {
         my $apr_ldflags=`$apr_config --ldflags`;
         chomp($apr_ldflags);
-        $opt .= " -rpath $CFG_LIBEXECDIR -module -avoid-version $apr_ldflags";
+        $opt .= " -rpath $CFG_LIBEXECDIR -Wl,--unresolved-symbols=ignore-all -module -avoid-version $apr_ldflags";
     }
 
     push(@cmds, "$libtool $ltflags --mode=link $CFG_CC -o $dso_file $opt $lo");
@@ -710,7 +710,7 @@
 
 -=#=-
 mod_%NAME%.la: mod_%NAME%.slo
-	$(SH_LINK) -rpath $(libexecdir) -module -avoid-version  mod_%NAME%.lo
+	$(SH_LINK) -rpath $(libexecdir) -Wl,--unresolved-symbols=ignore-all -module -avoid-version  mod_%NAME%.lo
 DISTCLEAN_TARGETS = modules.mk
 shared =  mod_%NAME%.la
 -=#=-
