--- httpd-2.4.6/configure.in.linux3.droplet	2013-07-19 23:16:56.966488049 +0200
+++ httpd-2.4.6/configure.in	2013-07-19 23:18:25.611915340 +0200
@@ -314,14 +314,7 @@ case $host in
       APR_SETVAR(SINGLE_LISTEN_UNSERIALIZED_ACCEPT, [1])
       ;;
   *-linux-*)
-      case `uname -r` in
-        # Unserialized accept() was not recommended until Linux 2.2.
-        [[01]].* | 2.[[01]]* )
-            ;;
-        * )
-            APR_SETVAR(SINGLE_LISTEN_UNSERIALIZED_ACCEPT, [1])
-            ;;
-      esac
+      APR_SETVAR(SINGLE_LISTEN_UNSERIALIZED_ACCEPT, [1])
       ;;
   *486-*-bsdi* | *-netbsd* | *-freebsd* | *-apple-darwin* | *-dec-osf* | *-qnx)
       APR_SETVAR(SINGLE_LISTEN_UNSERIALIZED_ACCEPT, [1])
